pkg/splunk/enterprise/indexercluster.go

```diff
@@ -226,7 +246,12 @@ func (mgr *indexerClusterPodManager) getClient(n int32) *splclient.SplunkClient
 
 // getClusterMasterClient for indexerClusterPodManager returns a SplunkClient for cluster master
 func (mgr *indexerClusterPodManager) getClusterMasterClient() *splclient.SplunkClient {
-	fqdnName := splcommon.GetServiceFQDN(mgr.cr.GetNamespace(), GetSplunkServiceName(SplunkClusterMaster, mgr.cr.GetName(), false))
+	masterIdxcName := mgr.cr.GetName()
+	// For multisite / multipart cluster, use the cluster-master API of the referenced IndexerCluster
+	if len(mgr.cr.Spec.IndexerClusterRef.Name) > 0 {
+		masterIdxcName = mgr.cr.Spec.IndexerClusterRef.Name
```

Comment:
Any reason not to allow cluster master to reside in a different namespace from indexer cluster CRs?

Reply:
HI Mike, thanks a lot for the review and for this finding. I'm rebasing the PR and fixing this issue.

Reply:
I actually found a possible reason not to allow this:
- The indexer service on the cluster master part would not select the indexers from the other parts (I didn't find any clear documentation, but for obvious security reasons I guess the selector of a Service only applies within a namespace)
- It would also not be possible to mount the secret from the cluster master part to the other IndexerCluster resources (documented [here](https://kubernetes.io/docs/concepts/configuration/secret/#restrictions)). This could be addressed by copying and synchronizing the secret, but because of the first limitation I'd like to get your feedback before working on this.

Then if referencing a cluster master residing in a different namespace is not supported, passing a namespace in IndexerClusterRef.Namespace in an IndexerCluster resource should probably be rejected with a clear error message.

Reply:
Makes sense. I think it's fine to just reject and say that crossing namespaces is not supported.